orttraining/orttraining/core/optimizer/graph_transformer_utils.cc

```diff
@@ -63,6 +60,13 @@
 #include "orttraining/core/optimizer/transformer_layer_recompute.h"
 #include "orttraining/core/optimizer/qdq_fusion.h"
 
+// Only enabled in full training build. Not in on device training builds
+#ifdef ENABLE_TRAINING
+#include "core/optimizer/compute_optimizer/upstream_gather.h"
+#include "core/optimizer/compute_optimizer/upstream_reshape.h"
+#include "orttraining/core/optimizer/compute_optimizer/sceloss_compute_optimization.h"
+#endif
```

Comment:
optional: Do we need to add the ifdef guard here if the underlying header files already have the guards?
I think it would be better if we just do it in one place.